cluster/dragon/dragon.go

```diff
@@ -997,39 +1001,64 @@ func (d *Dragon) tableExists(queryExec common.SimpleQueryExec, id uint64) (bool,
-// in the config after the cluster has been created. So we store the number in the database and check
-func (d *Dragon) checkConstantShards(expectedShards int) error {
-	log.Debugf("Checking constant shards: %d", expectedShards)
+func (d *Dragon) getConfigProperty(property string) ([]byte, error) {
 	key := table.EncodeTableKeyPrefix(common.LocalConfigTableID, 0, 16)
-	propKey := []byte("num-shards")
+	propKey := []byte(property)
 	key = common.AppendUint32ToBufferBE(key, uint32(len(propKey)))
 	key = append(key, propKey...)
-	v, err := d.LocalGet(key)
+	return d.LocalGet(key)
+}
+
+func (d *Dragon) setConfigProperty(property string, v []byte) error {
+	key := table.EncodeTableKeyPrefix(common.LocalConfigTableID, 0, 16)
+	propKey := []byte(property)
+	key = common.AppendUint32ToBufferBE(key, uint32(len(propKey)))
+	key = append(key, propKey...)
+	batch := d.pebble.NewBatch()
+	if err := batch.Set(key, v, nil); err != nil {
+		return errors.WithStack(err)
+	}
+	if err := d.pebble.Apply(batch, syncWriteOptions); err != nil {
+		return errors.WithStack(err)
+	}
+	return nil
+}
+
+func (d *Dragon) checkConstantProperty(property string, expectedValue int) error {
+	log.Debug("Checking constant %s: %d", property, expectedValue)
+	v, err := d.getConfigProperty(property)
 	if err != nil {
 		return err
 	}
 	if v == nil {
-		log.Debug("New cluster - no value for num-shards in storage, persisting it")
-		// New cluster - persist the value
-		v = common.AppendUint32ToBufferBE([]byte{}, uint32(expectedShards))
-		batch := d.pebble.NewBatch()
-		if err := batch.Set(key, v, nil); err != nil {
-			return errors.WithStack(err)
-		}
-		if err := d.pebble.Apply(batch, syncWriteOptions); err != nil {
-			return errors.WithStack(err)
+		log.Debugf("New cluster - no value for %s in stroage, persisting it", property)
+		v = common.AppendUint32ToBufferBE([]byte{}, uint32(expectedValue))
+		err = d.setConfigProperty(property, v)
+		if err != nil {
+			return err
 		}
 	} else {
-		shards, _ := common.ReadUint32FromBufferBE(v, 0)
-		log.Debugf("value for num-shards found in storage: %d", shards)
-		if int(shards) != expectedShards {
-			return errors.Errorf("number of shards in cluster cannot be changed after cluster creation. cluster value %d new value %d", expectedShards, shards)
+		value, _ := common.ReadUint32FromBufferBE(v, 0)
+		log.Debugf("value for %s found in storage: %d", property, value)
+		if int(value) != expectedValue {
+			return errors.Errorf("%s in cluster cannot be changed after cluster creation. cluster value %d new value %d", property, expectedValue, value)
 		}
 	}
 	return nil
 }
 
+// Currently the replication factor of the cluster is fixed. We must make sure the user doesn't change the replication factor
+// in the config after the cluster has been created. So we store the number in the database and check
+func (d *Dragon) CheckConstantReplicationFactor(expectedReplicationFactor int) error {
```

Comment:
No need for new functions here either. Just export the method 'checkConstantProperty'

Reply:
In fact, I would actually export set and get property instead - they are more useful

Reply:
> No need for new functions here either. Just export the method 'checkConstantProperty'

Do you mean to say, we can make use of the CheckConstantProperty(current, expected) directly to validate the properties instead of explicitly writing functions to check the individual properties?

Reply:
> In fact, I would actually export set and get property instead - they are more useful

So, export the get and set property functions and test them alone?

Reply:
Exported GetConfigProperty and SetConfigProperty. Added test as well.